tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -241,95 +315,98 @@ public void testCaseSensitiveEncryption() throws Exception {
     }
 
     @Test
-    public void testCreateAndLoadAndDeleteIdentity() throws Exception {
+    public void testCaseSensitive() throws Exception {
         FileSystemSecurityRealm securityRealm = new FileSystemSecurityRealm(getRootPath(), 3, ELYTRON_PASSWORD_PROVIDERS);
-        ModifiableRealmIdentity newIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
-        newIdentity.create();
-        newIdentity.dispose();
-
-        securityRealm = new FileSystemSecurityRealm(getRootPath(false), 3, ELYTRON_PASSWORD_PROVIDERS);
-        ModifiableRealmIdentity existingIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
-        assertTrue(existingIdentity.exists());
-        existingIdentity.delete();
-        assertFalse(existingIdentity.exists());
-        existingIdentity.dispose();
+        caseSensitive(securityRealm);
+    }
 
-        securityRealm = new FileSystemSecurityRealm(getRootPath(false), 3, ELYTRON_PASSWORD_PROVIDERS);
-        existingIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
-        assertFalse(existingIdentity.exists());
-        existingIdentity.dispose();
+    @Test
+    public void testCaseSensitiveEncryption() throws Exception {
+        FileSystemSecurityRealm securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath())
+                .setLevels(3)
+                .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                .setSecretKey(secretKey)
+                .build();
+        caseSensitive(securityRealm);
     }
 
     @Test
-    public void testCreateAndLoadAndDeleteIdentityEncryption() throws Exception {
+    public void testCaseSensitiveIntegrity() throws Exception {
         FileSystemSecurityRealm securityRealm = FileSystemSecurityRealm.builder()
                 .setRoot(getRootPath())
                 .setLevels(3)
-                .setSecretKey(key)
+                .setPrivateKey(privateKey)
+                .setPublicKey(publicKey)
                 .setProviders(ELYTRON_PASSWORD_PROVIDERS)
                 .build();
+        caseSensitive(securityRealm);
+    }
+
+    private void createAndLoadAndDeleteIdentity(String mode) throws Exception {
+        FileSystemSecurityRealmBuilder securityRealmBuilder = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath())
+                .setLevels(3)
+                .setProviders(ELYTRON_PASSWORD_PROVIDERS);
+        if (mode.equals("encryption")) { securityRealmBuilder.setSecretKey(secretKey); }
```

Comment:
Instead of this approach, it would be better to add `secretKey`, `privateKey`, and `publicKey` parameters to `createAndLoadAndDeleteIdentity` and then just set the builder methods if the values that are passed in aren't null.